### PR TITLE
Added withTimeLimiter and withThreadPoolBulkhead methods to Decorator…

### PR DESCRIPTION
**Pull request by RobWin**
Added withTimeLimiter, withThreadPoolBulkhead and withFallback methods to Decorators class.

### CODE REVIEW

**Comment by RobWin** [comment]
I had to do a change in the ThreadPoolBulkhead which is not backwards-compatible. But it should be ok, because it's more of a bugfix.

When a Runnable was submitted, the ThreadPoolBulkhead returned a Runnable instead of CompletionStage<Void>

**Comment by RobWin** [comment]
This PR allows to use the Decorators API as follows:

```
CompletableFuture<String> future = Decorators
    .ofSupplier(() -> helloWorldService.returnHelloWorld())
    .withThreadPoolBulkhead(ThreadPoolBulkhead.ofDefaults("helloBackend"))
    .withTimeLimiter(TimeLimiter.ofDefaults(), Executors.newSingleThreadScheduledExecutor())
    .withCircuitBreaker(circuitBreaker)
    .withFallback(TimeoutException.class, (e) -> "Recovery")
    .get().toCompletableFuture();
```

File: resilience4j-all/src/main/java/io/github/resilience4j/decorators/Decorators.java [hunk 1]
```diff
@@ -92,13 +102,29 @@ private DecorateSupplier(Supplier<T> supplier) {
...
+
         public Supplier<T> decorate() {
             return supplier;
         }
 
         public T get() {
             return supplier.get();
         }
+
+
```
Comment (dlsrb6342) [inline resilience4j-all/src/main/java/io/github/resilience4j/decorators/Decorators.java]: nit: could you remove empty lines?

**Comment by xlucasdemelo** [comment]
Why decorateRunnable does not return a Runnable anymore?

**Comment by Romeh** [comment]
> Why decorateRunnable does not return a Runnable anymore?
> 
> How can i execute a Supplier<CompletionStage> inside a ExecutorService ???

@xlucasdemelo  are you taking about Threadpool bulkhead for that question? if yes thread pool bulkhead has its own thread-pool executor service which be used to invoke ur runnable and return to completion stage for listentinh to the results of that invokation

**Pull request merged**